rbac/migration_tool/migrate.py

```diff
@@ -94,10 +97,52 @@ def migrate_role(
     return relationships, v2_role_bindings
 
 
-def migrate_users_for_groups(tenant: Tenant) -> list[common_pb2.Relationship]:
-    """Write users relationship to groups."""
+def migrate_system_role_assignments_for_groups(groups: list[Group]) -> list[common_pb2.Relationship]:
+    """Generate system role assignments for groups."""
+    with transaction.atomic():
```

Comment:
Is this required because of the select for update?

If not, I would probably take it out because it might imply it's needed or effective in some way that it isn't. Migration does NOT account for concurrent updates, so I am worried about implying that it does.

If it's required, maybe just add a comment to that effect (required due to select for update even though migrator does not generally account for concurrent updates and so should NOT be run outside of read-only mode.)

Reply:
yes it is and I wanted to ask what scope for transaction block should be in this case.

Reply:
It shouldn't really matter I guess since we shouldn't do this with concurrent writes. Not too big though, so a TX per group or per tenant is probably okay. Per group might be better but probably would require testing to confirm. (Normally there is a TX _per every single SQL operation_ so per group is not too many.)

Reply:
We also may want to add a check in the beginning of the migrator that requires read only mode is set otherwise it doesn't run as a safety mechanism.